fs/artifact_fetcher.go

```diff
@@ -177,24 +178,32 @@ func FetchSociArtifacts(ctx context.Context, refspec reference.Spec, indexDesc o
 		return nil, fmt.Errorf("cannot deserialize byte data to index: %w", err)
 	}
 
+	desc := ocispec.Descriptor{
+		Digest: indexDesc.Digest,
+		Size:   cw.Size(),
+	}
+
 	if !local {
 		b, err := soci.MarshalIndex(&index)
 		if err != nil {
 			return nil, err
 		}
-		err = localStore.Push(ctx, ocispec.Descriptor{
-			Digest: indexDesc.Digest,
-			Size:   cw.Size(),
-		}, bytes.NewReader(b))
+		err = localStore.Push(ctx, desc, bytes.NewReader(b))
 
 		if err != nil && !errors.Is(err, errdef.ErrAlreadyExists) {
 			return nil, fmt.Errorf("unable to store index in local store: %w", err)
 		}
+
+		err = store.LabelGCRoot(ctx, localStore, desc)
```

Comment:
Do we have a race if containerd cleans up the content between the `Push` and `LabelGCRoot`? I think this is what the lease service is for.

Reply:
We do have a lease, created when the content store is created. Per discussion I will confirm that the lease is dropped after the push+label and not held so long it expires.

Reply:
Latest push moves leases out to a new "Batch" concept which is currently directly analogous to a containerd lease, but I don't want to use that terminology to avoid committing too much to that implementation. A batch is created in each function that pushes content and labels it, and then expired at the end of that function.